packages/yarnpkg-pnp/sources/esm-loader/hooks/resolve.ts

```diff
@@ -18,12 +19,12 @@ export async function resolve(
     return defaultResolver(originalSpecifier, context, defaultResolver);
 
   let specifier = originalSpecifier;
-  const url = loaderUtils.tryParseURL(specifier);
+  const url = loaderUtils.tryParseURL(specifier, isRelativeRegexp.test(specifier) ? context.parentURL : undefined);
```

Comment:
Wouldn't it work to always specify the parent URL, even for absolute URLs?

Reply:
For absolute URLs it would just be ignored but for package names it would incorrectly resolve to a file
```js
console.log(new URL('no-deps', import.meta.url).href);
// file:///C:/berry/no-deps
```